bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -64,20 +65,24 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
             'newest': function(layer) {
                 me.getNewestLayers(20);
                 var ids = [];
-                jQuery(me._newestLayers).each(function(index, layer){
-                   ids.push(layer.getId());
+                jQuery(me._newestLayers).each(function(index, layer) {
+                    ids.push(layer.getId());
                 });
                 return (jQuery.inArray(layer.getId(), ids) !== -1);
             }
         };
+
+        this.layerlistFilterButtons = {};
```

Comment:
I don't think it's a good idea to gather UI-stuff in this service. Make a new service provided by layerselector bundle instead?